README.md

```diff
@@ -1,6 +1,147 @@
+  individual parameter sets. By default, all parameter sets are
+  included.
+- `rand`: Whereas the default APIs for `generate_key_pair` and
+  `encapsulate` expect external provision of random bytes, this
+  feature enables randomized versions of these APIs (in submodules
+  `mlkem512::rand`, `mlkem768::rand`, `mlkem1024::rand`) which take an
+  `&mut impl rand_core::CryptoRng` argument to generate the required
+  randomness internally.
+- `default-no-std` & `std`: Disabling default feature `std` provides
+  `no_std` support. For convenience `default-no-std` collects all default
+  features except `std`.
+  
+Additional features:
+- `kyber`: Provides access to an, as yet, unverified implementation of
+  Kyber as submitted in Round 3 of the NIST PQ competition. The Kyber
+  APIs follow the general structure of the ML-KEM APIs.
+- `check-secret-independence`: All operations on ring elements in the
+  portable implementation use the integer types of the
+  `libcrux-secrets` crate under the hood. That crate allows checking a
+  program operating on these types for secret indepence at compile
+  time. Enabling the `check-secret-independence` feature switches on
+  this compile-time checking of secret independence. By default, the
+  integer types of `libcrux-secrets` transparently fall back on Rust's
+  standard integer types.
+- `simd128`, `simd256`: These features force a compilation for NEON or
+  AVX2 targets, respectively, as discussed above.
+- `incremental`: An experimental API, which allows for incremental
+  encapsulation.
+
+## Security
+As outlined in the description of the `check-secret-independence`
+feature above, we leverage the Rust type system to ensure that secret
+values are not used in operations that are known to be
+non-constant time. While the implementation of constant time
+operations is best-effort, as there are no guarantees from the
+compiler, we follow established constant-time patterns and validate
+constant-time code via inspection of the generated assembly
+instructions.
+
+## Verification Status
+The portable and AVX2 code for field arithmetic, NTT polynomial
+arithmetic, serialization, and the generic code for high-level
+algorithms is formally verified using [hax](https://hax.cryspen.com) and [F*](https://fstar-lang.org).
+
+Please refer to [this
+file](https://github.com/cryspen/libcrux/blob/f9d1802c06ffe1dec900367e12b00edb1e7f9963/libcrux-ml-kem/proofs/verification_status.md)
+in the `libcrux-ml-kem` repository for details on the verification of
+this crate.
+
+## Performance
+We provide a dashboard of benchmark results for the main key
+generation, encapsulation and decapsulation APIs across different
+platforms and operating systems at
+[libcrux.cryspen.com](https://libcrux.cryspen.com).
+
+If you wish to run the benchmarks yourself, you can do so by running
+```
+cargo bench
+```
+from the crate root of the `libcrux-ml-kem` crate.
+
+## FAQ
+TBD
```

Comment:
I can't really think of anything here right now. Let's just drop it for now and add it when we have content.

```suggestion
```